File: mobile_api/schema_domains/proposal_entities.py
from __future__ import annotations

from ninja import Field, Schema


class ProposalKpisData(Schema):
    total_kcal: float | None = None
    protein: float | None = None
    carbs: float | None = None
    fat: float | None = None
    ppk: float | None = None
    alloc_protein: float | None = None
    alloc_carbs: float | None = None
    alloc_fat: float | None = None


class ProposalFoodData(Schema):
    food_id: int | None = None
    food_name: str
    quantity: float | None = None
    unit: str = "g"
    protein: float | None = None
    carbs: float | None = None
    fat: float | None = None
    total_kcal: float | None = None


class ProposalMealData(Schema):
    name: str
    foods: list[ProposalFoodData] = Field(default_factory=list)
    kpis: ProposalKpisData | None = None


class ProposalDailyPlanMealData(Schema):
    hour: str | None = None
    note: str = ""
    meal: ProposalMealData


class ProposalDailyPlanData(Schema):
    name: str
    meals: list[ProposalDailyPlanMealData] = Field(default_factory=list)
    kpis: ProposalKpisData | None = None

File: mobile/src/api/proposal-entities.ts
export type ProposalKpis = {
  total_kcal: number | null;
  protein: number | null;
  carbs: number | null;
  fat: number | null;
  ppk: number | null;
  alloc_protein: number | null;
  alloc_carbs: number | null;
  alloc_fat: number | null;
};

export type ProposalFood = {
  food_id: number | null;
  food_name: string;
  quantity: number | null;
  unit: string;
  protein: number | null;
  carbs: number | null;
  fat: number | null;
  total_kcal: number | null;
};

export type ProposalMeal = {
  name: string;
  foods: ProposalFood[];
  kpis: ProposalKpis | null;
};

export type ProposalDailyPlan = {
  name: string;
  meals: { hour: string | null; note: string; meal: ProposalMeal }[];
  kpis: ProposalKpis | null;
};

export type ProposalProgram = {
  name: string;
  duration_weeks: number;
  warnings: string[];
  days: { week_number: number; day_number: number; dailyplan: ProposalDailyPlan }[];
};
